### PR TITLE
Db transactions

### PR DESCRIPTION
**Pull request by jdorweiler**
This fixes two things:
- **Better error logging**  
 - If there is something wrong with the metadata file it logs the actual error so we know what happened

- **Transactions**
 - All database updates/deletes are in a single transaction.  This means if anything fails the whole transaction will roll back

@russellholt @jbarrett @MariagraziaAlastra

### CODE REVIEW

**Comment by MariagraziaAlastra** [comment]
@jdorweiler Nice!

File: script/iameta.pl [hunk 1]
```diff
@@ -30,114 +28,113 @@
...
-    #     print color 'reset';
-    # }
-    # exit 1 if (++$line > 5);
-
-
-}
+};
 
+try{
+    $d->db->txn_do( $update );
```
Comment (russellholt) [inline script/iameta.pl]: so does this commit for the entire operation?

Comment (jdorweiler) [inline script/iameta.pl]: yeah.  It's all or nothing.  Any errors will cause a rollback.

**Comment by russellholt** [comment]
LGTM

**Pull request merged**